PelotonEppSdk/Classes/PelotonClient.cs

```diff
@@ -1,27 +1,112 @@
             string stringResult;
             using (var client = new HttpClient())
             {
                 client.DefaultRequestHeaders.Authorization = content.authentication_header;
                 client.BaseAddress = factory.GetBaseUri();
                 var targetUriPart = factory.GetTargetUriPart(target);
-                var httpResponseMessage = await client.PostAsync(targetUriPart, stringContent);
+
+                HttpResponseMessage httpResponseMessage = null;
+                switch (type)
+                {
+                    case RequestType.Get:
+                        throw new NotImplementedException();
+                        break;
+                    case RequestType.Post:
+                		httpResponseMessage = await client.PostAsync(targetUriPart, stringContent);
+                        break;
+                    case RequestType.Put:
+                		httpResponseMessage = await client.PostAsync(targetUriPart, stringContent);
+                        break;
+                    case RequestType.Delete:
+                        httpResponseMessage = await client.DeleteAsync(targetUriPart);
+                        break;
+                    default:
+                        throw new NotImplementedException();
+                }
+
+                // handle server errors
+                if (!httpResponseMessage.IsSuccessStatusCode)
+                {
+                    throw new HttpException((int)httpResponseMessage.StatusCode, httpResponseMessage.ReasonPhrase);
+                }
+                stringResult = httpResponseMessage.Content.ReadAsStringAsync().Result;
+
+            }
+            return serializer.Deserialize<T>(stringResult);
+        }
+
+        /// <exception cref="HttpException">When status code is not <c>2XX Success</c>.</exception>
+        public async Task<T> PostAsync<T>(request_base content, ApiTarget target)
+        {
+            return await MakeBasicHttpRequest<T>(RequestType.Post, content, target);
+        }
+
+        /// <exception cref="HttpException">When status code is not <c>2XX Success</c>.</exception>
+        public async Task<T> PutAsync<T>(request_base content, ApiTarget target)
+        {
+            return await MakeBasicHttpRequest<T>(RequestType.Put, content, target);
+        }
+
+        /// <exception cref="HttpException">When status code is not <c>2XX Success</c>.</exception>
+        public async Task<T> DeleteAsync<T>(request_base content, ApiTarget target)
+        {
+            return await MakeBasicHttpRequest<T>(RequestType.Delete, content, target);
+        }
+
+        // Due to the nature of the BankAccounts Delete method, it must use this special Delete method
+        /// <exception cref="HttpException">When status code is not <c>2XX Success</c>.</exception>
+        public async Task<T> DeleteAsyncBankAccountsV1<T>(bank_account_delete_request content, ApiTarget target)
```

Comment:
Can this code be moved to the BankAccount object? Or does it have significant coupling to the PelotonClient class?